Core/src/org/sleuthkit/autopsy/coreutils/AppDBParserHelper.java

```diff
@@ -0,0 +1,1145 @@
+     */
+    public AppDBParserHelper(String moduleName, AbstractFile dbFile, Account.Type accountsType) throws TskCoreException {
+        
+        this.moduleName = moduleName;
+        this.dbAbstractFile = dbFile;
+        this.accountsType = accountsType;
+        this.selfAccountInstance = Case.getCurrentCase().getSleuthkitCase().getCommunicationsManager().createAccountFileInstance(Account.Type.DEVICE, ((DataSource)dbFile.getDataSource()).getDeviceId(), moduleName, dbFile);
+    }
+    
+    /**
+     * Constructs a AppDB parser helper for the given DB file.
+     * 
+     * This constructor is for Apps that do have app specific account information 
+     * for the device owner to create a 'self' account.
+     * 
+     * It creates a an account instance with specified type & id and uses it as
+     * a self account.
+     * 
+     * @param moduleName name module using the helper
+     * @param dbFile database file being parsed by the module
+     * @param accountsType account types created by this module
+     * @param selfAccountType self account type to be created for this module
+     * @param selfAccountId account unique id for the self account
+     * 
+     * @throws TskCoreException 
+     */
+    public AppDBParserHelper(String moduleName, AbstractFile dbFile, Account.Type accountsType, Account.Type selfAccountType, String selfAccountId) throws TskCoreException {
+        
+        this.moduleName = moduleName;
+        this.dbAbstractFile = dbFile;
+        this.accountsType = accountsType;
+        
+        this.selfAccountInstance = Case.getCurrentCase().getSleuthkitCase().getCommunicationsManager().createAccountFileInstance(selfAccountType, selfAccountId, moduleName, dbFile);
+    }
+        
+    /**
+     * Constructs a AppDB parser helper for the given DB file.
+     * 
+     * This is a constructor for Apps that do not need to create any 
+     * accounts/relationships.
+     * 
+     * @param moduleName name of module parsing the DB
+     * @param dbFile db file
+     * 
+     */
+    public AppDBParserHelper(String moduleName, AbstractFile dbFile) {
+        this.moduleName = moduleName;
+        this.dbAbstractFile = dbFile;
+        this.selfAccountInstance = null;
+        this.accountsType = null;
+    }
+    
+  
+    /**
+     * Creates and adds a TSK_CONTACT artifact to the case, with specified 
+     * attributes.
+     * Also creates an account instance of specified type for the contact with the 
+     * specified ID.
+     * 
+     * @param contactAccountUniqueID unique id for the contact's account
+     * @param contactName Name of contact
+     * @param phoneNumber primary phone number for contact
+     * @param homePhoneNumber home phone number 
+     * @param mobilePhoneNumber mobile phone number, 
+     * @param emailAddr Email address for contact
+     * 
+     * @return artifact created
+     * 
+     */
+    public BlackboardArtifact addContact(String contactAccountUniqueID, String contactName, 
```

Comment:
I have run into examples where a contact has multiple phone numbers, what do we do in this case?